apex-squid/src/test/java/org/fundacionjala/enforce/sonarqube/apex/parser/grammar/ApexGrammarQueryExpressionTest.java

```diff
@@ -0,0 +1,482 @@
+    }
+    
+    @Test
+    public void negativeRules_SimpleWhereExpression() {
+        assertThat(parser)
+                .notMatches("SELECT sub1 FROM AS LIMIT 200)")
+                .notMatches("SELECT field1,field2, (SELECT s FROM tab2 WHERE LIMIT 2) "
+                        + "FROM table1 WHEREcustomer=1 LIMIT 2000")
+                .notMatches("SELECT field1,field2 "
+                        + "FROM table1 WHERE customer_id = 'test'AST1 LIMIT 2000");
+    }
+    
+    @Test
+    public void positiveRules_ConditionalOperatorsWhereQueryExpression() {
+        assertThat(parser)
+                .matches("SELECT field1,field2 "
+                        + "FROM table1 AS T1 WHERE customer_id = 1 LIMIT 100")
+                .matches("SELECT field1,field2 "
+                        + "FROM table1 AS T1 WHERE Name LIKE 'A%' LIMIT 100")
+                .matches("SELECT field1,field2, "
+                        + "(SELECT sub1 FROM tab2 AS T2 WHERE id >= 1 LIMIT 2) "
+                        + "FROM table1 AS T1 WHERE customer_id = 'test' LIMIT 3000");
+    }
+    
+    @Test
+    public void negativeRules_ConditionalOperatorWhereQueryExpression() {
+        assertThat(parser)
+                .notMatches("SELECT sub1 FROM A WHEREcustomer=LIKE LIMIT 200)")
+                .notMatches("SELECT field1,field2, (SELECT s FROM tab2 WHERE LIMIT 2) "
+                        + "FROM table1 WHEREcustomer=1 LIMIT 2000")
+                .notMatches("SELECT field1,field2 "
+                        + "FROM table1 WHERE customer_id 'test' LIMIT 2000");
+    }
+    
+    @Test
+    public void positiveRules_ComplexWhereAndQueryExpression() {
+        assertThat(parser)
+                .matches("SELECT field1,field2 FROM table1 AS T1 "
+                        + "WHERE customer_id = 1 AND id = 1 LIMIT 100")
+                .matches("SELECT field1,field2 FROM table1 AS T1 "
+                        + "WHERE Name LIKE 'A%' AND C = 5 LIMIT 100")
+                .matches("SELECT field1,field2, "
+                        + "(SELECT sub1 FROM tab2 AS T2 WHERE id >= 1 LIMIT 2) "
+                        + "FROM table1 AS t1 WHERE Name LIKE 'A%' AND C = 5 "
+                        + "LIMIT 3000")
+                ;
+    }
+    
+    @Test
+    public void negativeRules_ComplexWhereAndQueryExpression() {
+        assertThat(parser)
+                .notMatches("SELECT field1,field2 FROM table1 AS T1 "
+                        + "WHERE customer_id = 1 ANDid = 1 LIMIT 100")
+                .notMatches("SELECT field1,field2 FROM table1 AS T1 "
+                        + "WHERE Name LIKE 'A%' ANDid = 'and' LIMIT 100")
+                .notMatches("SELECT field1,field2, "
+                        + "(SELECT sub1 FROM tab2 AS T2 WHERE id >= 1 LIMIT 2) "
+                        + "FROM table1 AS T1 "
+                        + "WHERE customer_id = 'test' Name LIKE 'A%' "
+                        + "AND ANDfield1 = 'Test' LIMIT 3000");
```

Comment:
I think SFDC can understand this statement without problems

AND ANDfield1 = 'Test' LIMIT 3000